### ISSUE TITLE
Ability to delete PMs from your inbox

### ISSUE THREAD

**Issue by DarrenMcRoy** [zooniverse/Panoptes-Front-End#1324]
Users will want to be able to delete private messages from their inbox to avoid it becoming overly cluttered, but they can't right now.

(Is there an issue for general PM inbox styling/functionality reform? I thought there was, but I can't find it.)

**Comment by parrish** [comment]
@aweiksnar If we can play around with this on staging for a bit, that'd probably be a good idea.  I'm pretty sure it should work alright, but I think it may be a bit weird.

Here's the weirdness:

- User1 has a conversation with User2
- User1 destroys the conversation (actually the user_conversation)
- User2 replies to the conversation
- User1 never receives the message (because they destroyed it)

I'm not sure if we want to inject some sort of (User1 left the conversation) sort of message automatically, or if we want to recreate the user conversation on further replies or what.

Any ideas?

**Comment by edpaget** [comment]
Would it work to allow users to just hide the conversation instead of actually deleting it? (ala archiving in gmail)

**Comment by aweiksnar** [comment]
I hadn't considered the case of the other user. Just to 'avoid clutter' maybe it just hides it from the user's inbox list? And then it becomes revived if the other user messages again, similar to an email chain. 

I'm not sure that we could let people 'leave' conversations, since user's could always just starter another conversation. It seems like that'd be a case for a larger user block/ban flag

**Comment by parrish** [comment]
Without getting too into implementation details, it's basically structured with a `User.has_many :conversations, through :user_conversations`.

So destroying a conversation for a user just destroys their user conversation.

If we want the conversation to reappear on another reply, then we could just recreate the user conversation on the message creation.

That said, it feels a bit weird to me to have the entire conversation just reappear after having "destroyed" it.  Maybe "hide" is a more appropriate label for the action.

**Comment by aweiksnar** [comment]
Yeah I think a 'hide' or 'watch' toggle would work be perfect for this

**Comment by DarrenMcRoy** [comment]
Are we still using "destroy" in any front-facing context? We've had feedback that people find that uncomfortable, and I sort of do, too. Every other place on the Internet uses "delete;" I've never seen "destroy," ever.

**Comment by aweiksnar** [comment]
@DarrenMcRoy Nope not currently

**Comment by vrooje** [comment]
I like "hide" or archive, instead of delete.

**Comment by yshish** [comment]
Just a question... I was testing the archive functionality.

 1. I sent a message from a student account to my yshish account. 
 2. Then I archived the message in the student mailbox (the message disappeared from the mailbox as expected).
 3. I went back to the yshish mailbox. The message in the yshish mailbox was displayed among other messages but there was no user name attached to the message, just the message subject. 

Is that an expected behaviour or an issue? Shouldn't be the message in yshish mailbox unaffected by that as it was archived on the other side only?
Screenshot (I blurred the other messages):

![screenshot_2017-07-02_13-22-59_archived-mail](https://user-images.githubusercontent.com/7211888/27769497-30039b4a-5f2b-11e7-8a72-3cdfc8940e0c.png)
